pkg/runtime/scheme.go

```diff
@@ -153,6 +153,38 @@ func (s *Scheme) AddUnversionedTypes(version unversioned.GroupVersion, types ...
 	}
 }
 
+func validateListType(t reflect.Type) {
+	// exceptions
+	if t.Name() == "APIGroupList" || t.Name() == "APIResourceList" {
```

Comment:
@smarterclayton, APIGroupList and APIResourceList don't comply with the old api-convention.md either. Can we live with these 2 exceptions? I think the schema validation is still useful to ensure future List kinds to conform to the rule.